src/graphql.rs

```diff
@@ -225,38 +247,33 @@ where
 
 const DEFAULT_CONNECTION_SIZE: i32 = 100;
 
-fn validate_pagination_params(
+fn default_connection_size(
```

Comment:
Given the name of this function, shouldn't it guarantee that one of the return values is `Some`? In the current implementation, it may be `(None, None)` if both `after` and `before` is `true`.

Reply:
I discovered that `clusters` and a few more APIs use `slicing::limit` to validate that both `first` and `last` cannot be used simultaneously (which is redundant) as well as to limit the connection size. Initially, I didn't consider limiting the size to avoid performance degradation. Based on what I've learned, I refactored the code to follow these criteria:

- If the connection size is not specified, provide a default size, except the case when `after` and `before` are both given. 
- If the connection size is specified but exceeds the limit, notify the users.
- Every API should use either `graphql::query` or `graphql::query_with_constraints`. The former relies on only `async_graphql::query` for pagination parameter validation, while the latter adds more constraints. Both ensure the above two points by calling `graphql::connection_size`